packages/core/src/ui/Base/TwitterEmbed.tsx

```diff
@@ -0,0 +1,124 @@
+import { useEffect, useCallback, useState, useRef, RefObject } from "react";
+
+const twitterWidgetJs = "https://platform.twitter.com/widgets.js";
+const message = "Loading tweet...";
+
+interface TweetConfig {
+  theme: string;
+}
+
+type TweetState = {
+  isLoading: boolean;
+  message: string | null;
+};
```

Comment:
Can you try using enum for all possible states instead? The message can later be derived based on the current Tweet state. But there is no need to keep it in state. Unless we would e.g. like to display some responses from the server. In our case we're only using "Loading..." or `null`.